scripts/bitrot.js

```diff
@@ -0,0 +1,243 @@
+
+let stylize = stylizeWithColor;
+
+// ---- support functions
+
+function rot(moduleName, s) {
+    gRotCount++;
+    console.log(`${stylize(moduleName, 'bold')} bitrot: ${s}`);
+}
+
+function getNpmInfo(name) {
+    const CACHE_TIMEOUT_MS = 30 * 60 * 1000; // 30 minutes
+    const cache = ensureCacheLoaded('npmInfo');
+    const cacheEntry = cache[name];
+    if (cacheEntry) {
+        if (cacheEntry.timestamp + CACHE_TIMEOUT_MS > Date.now()) {
+            return cacheEntry.value;
+        } else {
+            delete cache[name];
+        }
+    }
+
+    // Limited security guard on exec'ing given `name`.
+    const PKG_NAME_RE = /^(@[\w_.-]+\/)?([\w_.-]+)$/;
+    if (!PKG_NAME_RE.test(name)) {
+        throw new Error(
+            `${JSON.stringify(
+                name
+            )} does not look like a valid npm package name`
+        );
+    }
+
+    const stdout = execSync(`npm info -j "${name}"`);
+    const npmInfo = JSON.parse(stdout);
+
+    cache[name] = {
+        timestamp: Date.now(),
+        value: npmInfo,
+    };
+    saveCache();
+    return npmInfo;
+}
+
+function bitrot() {
+    const pj = JSON.parse(fs.readFileSync(ETEL_PJ_PATH, 'utf8'));
+    const instrNames = Object.keys(pj.dependencies).filter((d) =>
+        d.startsWith('@opentelemetry/instrumentation-')
+    );
+
+    const ainPj = getNpmInfo('@opentelemetry/auto-instrumentations-node');
+    const ainInstrNames = Object.keys(ainPj.dependencies).filter((d) =>
+        d.startsWith('@opentelemetry/instrumentation-')
+    );
+
+    for (let instrName of ainInstrNames) {
+        if (SKIP_INSTR_NAMES.includes(instrName)) continue;
+        if (!instrNames.includes(instrName)) {
+            rot(instrName, 'missing instr that auto-instrumentations-node has');
+        }
+    }
+
+    for (let instrName of instrNames) {
+        if (SKIP_INSTR_NAMES.includes(instrName)) continue;
+
+        if (!QUIET) console.log(`${instrName}:`);
+        const mod = require(instrName);
+        const instrClass = Object.keys(mod).filter((n) =>
+            n.endsWith('Instrumentation')
+        )[0];
+        const instr = new mod[instrClass]();
+        const initVal = instr.init(); // grpc is weird here
+        if (initVal === undefined) {
+            if (!QUIET) console.log(`    (instr.init() returned undefined!)`);
+            continue;
+        }
+        const instrNodeModuleFiles = Array.isArray(initVal)
+            ? initVal
+            : [initVal];
+
+        const supVersFromModName = {};
+        for (let inmf of instrNodeModuleFiles) {
+            // TODO: warn if supportedVersions range is open-ended. E.g. if it satisfies 9999.9999.9999 or something.
+            if (!QUIET)
+                console.log(
+                    `    ${inmf.name}: ${JSON.stringify(
+                        inmf.supportedVersions
+                    )}`
+                );
+            // TODO: keep printing these? Do they ever matter?
+            for (let file of inmf.files) {
```

Comment:
I don't see a use case for this yet. I'd say let's keep it if it's not too verbose